### PR TITLE
reverted changes that were removing reconnecting control port socket

### PR DESCRIPTION
**Pull request by ddeneka**

### CODE REVIEW

File: CPAProxy/CPAProxyManager.m [hunk 1]
```diff
@@ -164,7 +164,9 @@ - (void)tryConnectingControlPortAfterDelay:(NSTimeInterval)delay
 
 - (void)connectSocket
 {
-    self.controlPortStatus = CPAControlPortStatusConnecting;
+    if (self.controlPortStatus != CPAControlPortStatusConnecting) {
```
Comment (jaroslawgajak) [inline CPAProxy/CPAProxyManager.m]: shouldn't this be:
```
if (self.controlPortStatus == CPAControlPortStatusConnecting) {
return;
}
```
?

File: CPAProxy/CPAProxyManager.m [hunk 2]
```diff
@@ -231,7 +233,15 @@ - (void)socketManagerDidOpenSocket:(CPASocketManager *)manager
 - (void)socketManager:(CPASocketManager *)manager didDisconnectError:(NSError *)error
 {
     self.controlPortStatus = CPAControlPortStatusClosed;
-    // Do not try reconnecting. We restart the whole thing anyways
+    self.controlPortConnectionAttempts += 1;
+    if(self.controlPortConnectionAttempts < CPAMaxNumberControlConnectionAttempts) {
+        self.controlPortStatus = CPAControlPortStatusConnecting;
```
Comment (jaroslawgajak) [inline CPAProxy/CPAProxyManager.m:223]: Why setting controlPortStatus here, not in connect method?

File: CPAProxy/CPAProxyManager.m (hunk 1 above)
Comment (jaroslawgajak) [reply CPAProxy/CPAProxyManager.m]: Why setting controlPortStatus to CPAControlPortStatusConnecting removed from here?

**Comment by lukewar** [comment]
@ddeneka-ynd why did we reverted those changes?

**Review by lukewar — APPROVED** [review]

**Pull request merged**